swift/apple/FirezoneNetworkExtension/NetworkSettings.swift

```diff
@@ -166,6 +154,23 @@ class NetworkSettings {
       completionHandler?(error)
     }
   }
+
+  // See https://github.com/firezone/firezone/issues/2939
+  // 1. Read existing system default servers from resolv.conf
+  // 2. Remove our sentinel if it's already there
+  // 3. Create a new Array with our sentinel first (order matters)
+  private func createDNSSettings(dnsSentinelAddress: String) -> NEDNSSettings {
+    var systemDefaultResolvers = Resolv().getservers().map(Resolv.getnameinfo)
+    systemDefaultResolvers.removeAll(where: { dnsSentinelAddress == $0 })
+    let dnsServers = [dnsSentinelAddress] + systemDefaultResolvers
+    let dnsSettings = NEDNSSettings(servers: dnsServers)
```

Comment:
So if I understand correctly, at this point we can get the actual system's DNS servers and this is why this works?

Then, instead of adding them to the tunnel's dns setting could we just store them and have `getSystemsDefaultResolvers` just return the stored settings here?

We could use an static variable or better just some field that we keep around in the tunnel state.

Reply:
@conectado Yeah that's correct, and probably the better approach.

@roop Was chatting with @conectado about this and it seems like what we should do for now is just save the dns servers before starting the tunnel so that connlib always has the tight dns servers returned whenever it calls getSystemDefaultResolvers